beacon_node/beacon_chain/src/light_client_server_cache.rs

```diff
@@ -119,23 +128,55 @@ impl<T: BeaconChainTypes> LightClientServerCache<T> {
             )?);
         };
 
+        let maybe_finalized_block_root =
+            store.get_full_block(&cached_parts.finalized_block_root)?;
+
+        let new_light_client_update = LightClientUpdate::new(
+            sync_aggregate,
+            block_slot,
+            cached_parts.next_sync_committee,
+            cached_parts.next_sync_committee_branch,
+            cached_parts.finality_branch.clone(),
+            &attested_block,
+            maybe_finalized_block_root.as_ref(),
+            chain_spec,
+        )?;
+
+        let sync_period = block_slot
+            .epoch(T::EthSpec::slots_per_epoch())
+            .sync_committee_period(chain_spec)?;
+
+        let prev_light_client_update =
+            self.get_light_client_update(&store, sync_period, chain_spec)?;
```

Comment:
As an optimization, you can cache a single item from the store to prevent reading the same update over and over.